overrides/default.nix

```diff
@@ -881,11 +881,11 @@ lib.composeManyExtensions [
 
       fastapi = prev.fastapi.overridePythonAttrs (old: {
         # fastapi 0.111 depends on fastapi-cli, which depends on fastapi, resulting in infinite recursion
-        propagatedBuildInputs = builtins.filter (e: e.pname != "fastapi-cli") old.propagatedBuildInputs;
+        propagatedBuildInputs = removePackagesByName (old.propagatedBuildInputs or [ ]) [ final.fastapi-cli ];
```

Comment:
`final.fastapi-cli` exists only for `fastapi` version 0.111.0. The previous versions of `fastapi` do not depend on `final.fastapi-cli`.

Reply:
```suggestion
        propagatedBuildInputs = removePackagesByName (old.propagatedBuildInputs or [ ]) lib.optional final?fastapi-cli final.fastapi-cli;
```